diff -Naur inetsim.orig/lib/INetSim/DNS.pm inetsim/lib/INetSim/DNS.pm
--- inetsim.orig/lib/INetSim/DNS.pm	2017-03-11 18:28:35.029725577 +0100
+++ inetsim/lib/INetSim/DNS.pm	2017-03-11 18:28:43.679725064 +0100
@@ -168,19 +168,21 @@
         my $rdata;
         # http://www.ietf.org/rfc/rfc4892.txt
         # http://www.ietf.org/proceedings/54/I-D/draft-ietf-dnsop-serverid-00.txt
-        if ($queryclass eq "CH" && ($queryname =~ /^(version|hostname)\.bind/i || $queryname =~ /^(id|version)\.server/i)) {
+        if ($queryname =~ /^[0-9a-zA-Z-.]{1,255}$/) {
+            if ($queryclass eq "CH" && ($queryname =~ /^(version|hostname)\.bind/i || $queryname =~ /^(id|version)\.server/i)) {
             $rdata = &INetSim::Config::getConfigParameter("DNS_Version");
         }
-        elsif ($queryname =~ /^[0-9a-zA-Z-.]{1,255}$/) {
+        else {
             $rdata = "this is a txt record";
-	    push @ans, Net::DNS::RR->new("$queryname $ttl $queryclass TXT \"$rdata\"");
-	    push @logans, "$queryname $ttl $queryclass $querytype \"$rdata\"";
-	    $resultcode = "NOERROR";
         }
-	else {
-	    # invalid queryname
-	    $resultcode = "NXDOMAIN";
-	}
+        push @ans, Net::DNS::RR->new("$queryname $ttl $queryclass TXT \"$rdata\"");
+        push @logans, "$queryname $ttl $queryclass $querytype \"$rdata\"";
+        $resultcode = "NOERROR";
+        }
+    else {
+        # invalid queryname
+        $resultcode = "NXDOMAIN";
+    }
     }
 
     elsif ($querytype eq "MX") {
