diff -Naur opt/inetsim.orig/lib/INetSim/DNS.pm opt/inetsim/lib/INetSim/DNS.pm
--- opt/inetsim.orig/lib/INetSim/DNS.pm	2016-08-29 09:43:28.000000000 +0200
+++ opt/inetsim/lib/INetSim/DNS.pm	2017-03-08 22:45:46.297547205 +0100
@@ -30,11 +30,13 @@
     my $localaddr = (defined &INetSim::Config::getConfigParameter("DNS_BindAddress") ? &INetSim::Config::getConfigParameter("DNS_BindAddress") : &INetSim::Config::getConfigParameter("Default_BindAddress"));
     $localaddr =~ /^(.*)$/;  # fool taint check
     $localaddr = $1;
+    my $localport = (defined &INetSim::Config::getConfigParameter("DNS_BindPort") ? &INetSim::Config::getConfigParameter("DNS_BindPort") : &INetSim::Config::getConfigParameter("Default_BindPort"));
+    $localport =~ /^(.*)$/; $localport = $1;
     local $SIG{'INT'} = 'IGNORE';
     local $SIG{'TERM'} = sub {&INetSim::Log::MainLog("stopped (PID $CPID)", &INetSim::Config::getConfigParameter("DNS_ServiceName")); exit 0;};
 
     my $server = Net::DNS::Nameserver->new(LocalAddr    => $localaddr,
-					   LocalPort    => &INetSim::Config::getConfigParameter("DNS_BindPort"),
+					   LocalPort    => $localport,
 					   ReplyHandler => \&dns_reply_handler,
 					   Verbose      => '0');
     if(! $server) {
